src/radical/entk/appman/appmanager.py

```diff
@@ -481,44 +466,27 @@ def run(self):
             self._logger.exception('Error in AppManager: %s' % ex)
 
             # Terminate threads in following order: wfp, helper, synchronizer
-            if self._wfp:
-                self._logger.info('Terminating WFprocessor')
-                self._wfp.terminate_processor()
-
-            if self._task_manager:
-                self._logger.info('Terminating task manager process')
-                self._task_manager.terminate_manager()
-                self._task_manager.terminate_heartbeat()
-
-            if self._sync_thread:
-                self._logger.info('Terminating synchronizer thread')
-                self._terminate_sync.set()
-                self._sync_thread.join()
-                self._logger.info('Synchronizer thread terminated')
-
-            if self._resource_manager:
-                self._resource_manager._terminate_resource_request()
+            
+            self._terminate()
 
             self._prof.prof('termination done', uid=self._uid)
             raise
 
-    def resource_terminate(self):
+    def terminate(self):
 
-        if self._task_manager:
-            self._logger.info('Terminating task manager process')
-            self._task_manager.terminate_manager()
-            self._task_manager.terminate_heartbeat()
-
-        if self._resource_manager:
-            self._resource_manager._terminate_resource_request()
+        self._prof.prof('start termination', uid=self._uid)
+        self._terminate()
+        self._prof.prof('termination done', uid=self._uid)
 
-        if os.environ.get('RADICAL_ENTK_PROFILE', False):
-            write_session_description(self)
 
-        if self._rmq_cleanup:
-            self._cleanup_mqs()
+    def resource_terminate(self):
+        
+        self._prof.prof('start termination', uid=self._uid)
+        self._logger.warning('DeprecationWarning: Public Method resource_terminate is deprecated.\
+                             Please use terminate')
+        self._terminate()
+        self._prof.prof('termination done', uid=self._uid)
```

Comment:
Just use `self.temrinate()` instead of repeating terminate's code?

Reply:
I introduced `_terminate` because that is what we said I will do, as it is written in issue #330. I can change it if everyone is in favor of keeping only the public method.